app/models/user.rb

```diff
@@ -9,4 +9,11 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :friends, -> { where(friendships: { status: 'accepted' }).order('name DESC') },
+           through: :friendships
+  has_many :requested_friends, -> { where(friendships: { status: 'requested' }).order('created_at DESC') },
+           through: :friendships, source: :friend
+  has_many :pending_friends, -> { where(friendships: { status: 'pending' }).order('created_at DESC') },
+           through: :friendships, source: :friend
 end
```

Comment:
You should add a method here called something like
`friends_and_own_posts` that returns posts of the user and user's friends which can then be called on the timeline.
A simple `User.where` where use if the user (self) or `user.friends` should suffice